src/utils/EventUtils.ts

```diff
@@ -58,6 +59,13 @@ export function spreadEvent(args: Result | Record<string, unknown>): { [key: str
   if (returnedObject.rootBundleId) {
     returnedObject.rootBundleId = Number(returnedObject.rootBundleId);
   }
+  // If depositId is included in the event, cast it to a BigNumber. Need to check if it is defined since the deposit ID can
+  // be 0, which would still make this evaluate as false.
+  if (isDefined(returnedObject.depositId)) {
+    // Assuming a numeric output, we can safely cast the unknown to BigNumberish since the depositId will either be a uint32 (and therefore a TypeScript `number`),
+    // or a uint256 (and therefore an ethers BigNumber).
+    returnedObject.depositId = toBN(returnedObject.depositId as BigNumberish);
+  }
```

Comment:
Were you able to confirm what is natively returned here? I guess it looks like a `number` by default, since we don't already check for it here. With the new ABI then I'd assume ethers is storing it as a BigNumber instead. In that case we might subsequently be able to drop this check again, once we're over the upgrade hurdle.

Reply:
Yes. A `number` is returned here. We can/should delete this part once the upgrade rolls out.